tox/venv.py

```diff
@@ -149,9 +150,27 @@ def is_allowed_external(self, p):
                     return True
         return False
 
+    def _real_python3(self, python):
+        """ use real_prefix to determine if we're running inside a virtualenv,
+            and if so, use it as the base path to determine the real python
+            executable path.
+        """
+        args = [str(python), '-c', 'import sys; print(sys.real_prefix)']
+
+        process = subprocess.Popen(args, stdout=subprocess.PIPE)
```

Comment:
I've opted not to use `action.popen()` here, given that it causes a lot of tests to fail (test are expecting a single popen call, but this would create a second call, so the count assertions fail).